### PR TITLE
[PLAYER-556] Better parsing of URL parameters

### PR DESCRIPTION
**Pull request by mityaha**

### CODE REVIEW

File: amp_iframe.html [hunk 1]
```diff
@@ -12,39 +12,50 @@
...
-    }
+      var nvp = query.split('&');
+      var data = {};
+      for (var i = 0 ; i < nvp.length ; i++) {
+        var pair = nvp[i].split('=');
+        var name = decodeURIComponent(pair[0]);
+        var value = decodeURIComponent(pair[1]);
+
+        var path = name.match(/(^[^\[]+)(\[.*\]$)?/);
+        var first = path[1];
```
Comment (pilievOoyala) [inline amp_iframe.html]: what happens if path contains 0 elements? Need a check.

File: iframe.html [hunk 2]
```diff
@@ -13,39 +13,50 @@
...
+
+      var nvp = query.split('&');
+      var data = {};
+      for (var i = 0 ; i < nvp.length ; i++) {
+        var pair = nvp[i].split('=');
+        var name = decodeURIComponent(pair[0]);
+        var value = decodeURIComponent(pair[1]);
+
+        var path = name.match(/(^[^\[]+)(\[.*\]$)?/);
+        var first = path[1];
```
Comment (pilievOoyala) [inline iframe.html]: Need to check that path actually contains elements here as well.

Comment (pilievOoyala) [inline iframe.html]: Don't we want to start with the 0th element, not 1?

Comment (mityaha) [inline iframe.html]: key-value: key is element #0 of pair, value is element #1, value may be encoded, so we want to decode it, so we deal with element #1 here, element #0 doesn't require decoding

File: amp_iframe.html (hunk 1 above)
Comment (mityaha) [inline amp_iframe.html]: Added

File: iframe.html (hunk 2 above)
Comment (mityaha) [inline iframe.html]: No, zeroth element returns whole expression, matching parts start with index 1

File: amp_iframe.html [hunk 3]
```diff
@@ -12,39 +12,56 @@
...
-        } 
-
-        if (result[key]) {
-          if (Array.isArray(result[key])) {
-            result[key].push(value);
-          } else {
-            result[key] = [result[key], value];
+    function queryStringToJSON(query) {
+      var setValue = function(root, path, value) {
+        if (path.length > 1){
```
Comment (pilievOoyala) [inline amp_iframe.html]: I think it needs to be path && path.length.  What is path is null?

Comment (mityaha) [inline amp_iframe.html]: Path has already been verified by code that calls setValue, this is actually redundant check altogether.

**Comment by pilievOoyala** [comment]
LGTM

**Pull request merged**